### DISCUSSION TITLE
OCR - Should image file accepted when option is on ?

### DISCUSSION THREAD

**Discussion by leobenkel** [open-webui/open-webui#2466] [General]
**Is your feature request related to a problem? Please describe.**
Even when the "OCR" option is turned on, image files are not accepted

**Describe the solution you'd like**
When the OCR option is ON, then image files should be accepted.

**Comment by drupol** [comment]
OCR are made for PDFs.

**Comment by mistertwo** [comment]
It looks like openCV [is being pulled in as a dependency](https://github.com/open-webui/open-webui/blob/24b638fcdea17fb8621e0391150bee852cf74a38/backend/requirements.txt#L52), and it has OCR capabilities. I'll take a look here soon to see if something can be cobbled together to support OCR for images.

EDIT: Looks like OpenCV would be part of the process, but not handle the OCR itself. Still going to see if there's something useful here.

**Reply by mistertwo** [reply]
So, after playing with OpenCV and Tesseract for a bit, along with experimenting with LlaVa, I don't think this is going to be fun. :)

What may move things forward could be some functionality that passes off file payloads to an external service for analysis (based on mime type), and that external service returns the text from the file payload for processing by the chosen LLM.

**Reply by linbanana** [reply]
Maybe can convert the image to a PDF file first, and then use the OCR functionality of the PDF to recognize the text from the image.
